lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -694,6 +698,28 @@ instance FromText PoolId where
             , "bytes in length."
             ]
 
+-- | Encode 'PoolId' as Bech32 with "pool" hrp.
+encodePoolIdBech32 :: PoolId -> T.Text
+encodePoolIdBech32 =
+    Bech32.encodeLenient hrp
+        . Bech32.dataPartFromBytes
+        . getPoolId
+      where
+        hrp = [Bech32.humanReadablePart|pool|]
+
+-- | Decode a Bech32 encoded 'PoolId'. Throws 'TextDecodingError' on failure.
+decodePoolIdBech32 :: MonadThrow m => T.Text -> m PoolId
```

Comment:
**Question:** is there an advantage to using `MonadThrow` over `Either` here?

**Suggestion:** use `Either`

```hs
-- | Decode a Bech32-encoded 'PoolId'.
decodePoolIdBech32 :: T.Text -> Either TextDecodingError PoolId
```
**Advantages**:
* the type of failure is made explicit in the type signature
* this type signature is consistent with other existing decoders (for example `hashFromText`).
* the comment can be simpler.

However, I do wonder if there is some advantage to using `MonadThrow` instead?

Reply:
> Question: is there an advantage to using MonadThrow over Either here?

MonadThrow specialises to Either, Maybe and IOException in IO, so has a richer API. The downside is that `e ~ SomeException`. It is easy to change it, but that was my decision so far.